### PR TITLE
Add debian files

### PR DESCRIPTION
**Pull request by b4ldr**
creating to foster comments

built using
```
$ gbp buildpackage --git-debian-branch=debian -us -uc
```
@massimocandela just in case you see this, will send a PR once its cleaned up, howeever it will not be a debian standard package as im vendoring in all dependencies

### CODE REVIEW

File: debian/rules [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+	cd "${CURDIR}/build" && npm pack
+
+override_dh_auto_test:
+	/bin/true
+
+override_dh_auto_install:
+	npm install --prefix "${CURDIR}/debian/node-bgpalerter/usr" -g ${CURDIR}/build/bgpalerter-*.tgz
+	find "${CURDIR}/debian/node-bgpalerter/usr/lib/node_modules/bgpalerter/node_modules" \
+		-name .npmignore -name .eslintrc -name \*.md -name LICENSE -name LICENSE-jsbn -name package.json \
+		-name .gitmodules -name .gitattributes -delete
```
Comment (b4ldr) [inline debian/rules:19]: this find command dosn;t appear to run

File: debian/rules [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+
+override_dh_auto_link:
+	# We dont want to link the vendored packages
+	find "${CURDIR}/debian/node-bgpalerter/usr/lib/node_modules/bgpalerter/node_modules" \
+		-name examples -name .bin -name bin -type d -delete
```
Comment (b4ldr) [inline debian/rules:24]: dh_link is still running and i think breaking the permissions on /usr/lib/node_modules/bgpalerter/index.js

**Comment by massimocandela** [comment]
also, if you run "npm run compile", in `dist/` you will find the "compiled" version, which runs without `devDependencies` (they don't even need to be installed at that point). Not sure if this info is useful

**Comment by b4ldr** [comment]
>Can the code in build.sh be useful somehow? it creates a self-contained binary, which for sure cannot be proposed as official deb package, but if that is not the idea in the first place maybe it can be used.

I spoke with a colleague internally and they suggested that its better to have the files vendored in as directory in the files system, i guess this is so users can manually apply fixes/security patches etc.  As well as review what is actually being installed.   its a middle ground between being Debian compliment and packaging a blackbox binary

FYI once i have this all cleaned up all also spin of https://github.com/wikimedia/puppet/tree/production/modules/bgpalerter and add it to forge

> if you run "npm run compile", in dist/ you will find the "compiled" version
we do the following in the debian/rules file, which AFAIK (however i dont know much when it comes to node :)) is the same as `npm run compile`.  running npm compile didn't work out of the box as babel is named babeljs in debian

```makefile
override_dh_auto_build:
	babeljs index.js -d "${CURDIR}/build"
	babeljs src -d "${CURDIR}/build/src"
	cp package.json "${CURDIR}/build/"
```

**Pull request closed without merging**